api/base/parsers.py

```diff
@@ -182,7 +182,7 @@ def parse(self, stream, media_type=None, parser_context=None):
             if id_ is None:
                 raise JSONAPIException(source={'pointer': '/data/id'}, detail=NO_ID_ERROR)
 
-            if type_required and parser_context['request'].version >= 2.7 and request_method == 'PATCH':
+            if type_required and request_method == 'PATCH':
```

Comment:
`type` is also required for `POST` requests:

> A resource can be created by sending a POST request to a URL that represents a collection of resources. The request MUST include a single resource object as primary data. The resource object MUST contain at least a type member.

http://jsonapi.org/format/#crud-creating

Reply:
solved in 2ab5eed